Library/Homebrew/utils/shell.rb

```diff
@@ -40,14 +40,31 @@ def profile
       SHELL_PROFILE_MAP.fetch(preferred, "~/.bash_profile")
     end
 
+    def prepend_set_variable_in_profile(variable, value)
+      case preferred
+      when :bash, :ksh, :sh, :zsh, nil
+        "echo 'export #{variable}=#{sh_quote(value)}${#{variable}+:\"$#{variable}\"}' >> #{profile}"
+      when :csh, :tcsh
+        <<~EOS.chomp!
+          if ( $?#{variable} == 0 )
+            setenv #{variable} #{csh_quote(value)} >> #{profile}
+          else
+            setenv #{variable} #{csh_quote(value)}:"$#{variable}" >> #{profile}
+          endif
+        EOS
+      when :fish
+        "echo 'set -gx #{variable} #{sh_quote(value)} \"$#{variable}\"' >> #{profile}"
+      end
+    end
+
     def prepend_path_in_profile(path)
       case preferred
       when :bash, :ksh, :sh, :zsh, nil
-        "echo 'export PATH=\"#{sh_quote(path)}:$PATH\"' >> #{profile}"
```

Comment:
Can you explain all the changes in this function and why they are needed?

Reply:
I'm a bit dubious that this is all completely broken but yet we've not handle a single related used quote. Can you extract all the changes to this function (and the equivalents for `prepend_set_variable_in_profile`) into a separate PR where it can be discussed there. For example:

> export PATH="~/test test:$PATH"

This seems vastly preferable to:

> export PATH=~/test\ test:"$PATH"

Reply:
OK, I will move it to a different PR.

Reply:
Thanks!

Reply:
It may be preferable, but not what Homebrew does. Homebrew says:

> export PATH="~/test\ test:$PATH"

Reply:
> `~` must be outside of the quotes.

exactly. When it's outside of double quotes, it is correctly translated to user's home directory